src/Data/Parameterized/Context/Safe.hs

```diff
@@ -655,6 +665,7 @@ traverseWithIndex :: Applicative m
                   -> Assignment f ctx
                   -> m (Assignment g ctx)
 traverseWithIndex f a = generateM (size a) $ \i -> f i (a ! i)
+{-# DEPRECATED traverseWithIndex "Use itraverseFC" #-}
```

Comment:
I'm perhaps not completely up-to-date on the conventions employed in `parameterized-utils`, but does it make sense to all-out deprecate functions like `traverseWithIndex`? Other parts of `parameterized-utils` happily define specialized versions of class methods (e.g., [`Data.Parameterized.Map.map`](https://hackage.haskell.org/package/parameterized-utils-2.1.3.0/docs/Data-Parameterized-Map.html#v:map)).

Reply:
Honestly, I have no particular opinion on this - My instinct was that it made sense to have just one name for these things, but I'd be happy to keep around the specializations.

Reply:
Considering that `traverseWithIndex` is fairly commonly used various projects (see [here](https://github.com/search?q=%22traverseWithIndex%22+user%3AGaloisInc&type=code)), my inclination is to avoid the code churn that a deprecation cycle would induce.

Reply:
I replaced the deprecations with notes in the Haddocks. Thanks for the note, I really hadn't considered the impact on downstream projects.